dwa_local_planner/src/dwa_planner_ros.cpp

```diff
@@ -483,5 +485,19 @@ namespace dwa_local_planner {
 
   }
 
+  void DWAPlannerROS::updateCostmap(costmap_2d::Costmap2D* master) {
+    boost::unique_lock<costmap_2d::Costmap2D::mutex_t> lock(*(master->getMutex()));
+    if (costmap_copy_.getSizeInCellsX() != master->getSizeInCellsX() ||
+        costmap_copy_.getSizeInCellsY() != master->getSizeInCellsY() ||
+        costmap_copy_.getResolution() != master->getResolution()) {
+       costmap_copy_.resizeMap(master->getSizeInCellsX(), master->getSizeInCellsY(), master->getResolution(),
+                               master->getOriginX(), master->getOriginY());
+    }
+    else {
+       costmap_copy_.updateOrigin(master->getOriginX(), master->getOriginY());
+    }
+    size_t mem_size = costmap_copy_.getSizeInCellsX() * costmap_copy_.getSizeInCellsY() * sizeof(unsigned char);
+    std::memcpy(costmap_copy_.getCharMap(), master->getCharMap(), mem_size);
```

Comment:
```suggestion
    boost::unique_lock<costmap_2d::Costmap2D::mutex_t> lock(*(master->getMutex()));
    costmap_copy_ = master;
```
S: if you're going to copy the entire costmap with memcpy then might as well just assign it

Reply:
But then copied pointer will keep pointing to the original cost array, as it has no copy constructor defined

Reply:
What do you mean? :thinking:  Costmap2D overloads the assignment operator and it does a memcpy on the charmap, just like you're doing in your code.

https://github.com/rapyuta-robotics/ros-navigation/blob/48a403ac942878a2d11ef756c8b3d17fa651ca2b/costmap_2d/src/costmap_2d.cpp#L137-L159

Reply:
True,,,  changed